### PR TITLE
storagecluster: rewrite cephblockpool reconcilation

### PR DESCRIPTION
**Pull request by rewantsoni**
The current implementation assumes only a single entity can touch the CR. Hence modifying the implementation such that the controller owns the specific fields and other entities are allowed to modify other fields of the cephblockpool CR.

Part of [RHSTOR-4886](https://issues.redhat.com/browse/RHSTOR-4886)

### CODE REVIEW

File: controllers/storagecluster/cephblockpools.go [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 type ocsCephBlockPools struct{}
 
 // ensures that peer cluster secret exists and adds it to CephBlockPool
-func (r *StorageClusterReconciler) addPeerSecretsToCephBlockPool(initData *ocsv1.StorageCluster, poolName string, poolNamespace string) cephv1.MirroringPeerSpec {
+func (o *ocsCephBlockPools) addPeerSecretsToCephBlockPool(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, poolName string, poolNamespace string) cephv1.MirroringPeerSpec {
```
Comment (rchikatw) [inline controllers/storagecluster/cephblockpools.go:21]: signature can also be modified to this I guess addPeerSecretsToCephBlockPool(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, `poolName, poolNamespace string`)

File: controllers/storagecluster/cephblockpools.go [hunk 2]
```diff
@@ -39,199 +40,276 @@ func (r *StorageClusterReconciler) addPeerSecretsToCephBlockPool(initData *ocsv1
...
-				},
-			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      builtinMgrPoolName,
-				Namespace: poolNamespace,
-			},
-			Spec: cephv1.NamedBlockPoolSpec{
-				Name: ".mgr",
```
Comment (iamniting) [inline controllers/storagecluster/cephblockpools.go:80]: This name is missed from the new code.

**Review by iamniting — CHANGES_REQUESTED** [review]

**Review by iamniting — APPROVED** [review]
The PR looks good to me. Can you pls test it once?

/lgtm
/hold for testing

**Review by iamniting — APPROVED** [review]

**Comment by rewantsoni** [comment]
Editing of the cephblockpool mirroring field manually is allowed even if mirroring is not enabled from storageCluster CR
```
[rewantsoni scripts] k get storagecluster ocs-storagecluster -oyaml | yq '.spec.mirroring'
{}

[rewantsoni scripts] k get cephblockpool ocs-storagecluster-cephblockpool -oyaml | yq '.spec.mirroring'
enabled: true
mode: image
```

**Pull request merged**